### DISCUSSION TITLE
hover_swarm.launch throwing 'std::runtime_error'   what():  timeout.

### DISCUSSION THREAD

**Discussion by aarsht7** [USC-ACTLab/crazyswarm#623] [Q&A] [answered]
Hello, I am trying to configure the crazyswarm. All the previous steps of configuring went well. When I tried to do `roslaunch crazyswarm hover_swarm.launch` it gives following error.

Note that I have configured the crazyflie to number 4 as upto number 3 was used by other CFs. I have made the required changes in the configuration files as well.

`python3 hello_world.py --sim` is working properly.

`mocap_helper.launch` is giving me good output of the position of the CF.
 here is my [hover_swarm.launch](https://pastebin.com/MZjqwEF2).
 here is my [crazyflies.yaml](https://pastebin.com/pwFRRHrs)

```
A long list of parameters here that I have skipped

NODES
  /
    crazyswarm_server (crazyswarm/crazyswarm_server)
    crazyswarm_teleop (crazyswarm/crazyswarm_teleop)
    joy (joy/joy_node)

auto-starting new master
process[master]: started with pid [22216]
ROS_MASTER_URI=http://localhost:11311

setting /run_id to 5f8c5dc8-b658-11ec-a67f-5d216f884988
process[rosout-1]: started with pid [22241]
started core service [/rosout]
process[crazyswarm_server-2]: started with pid [22248]
process[joy-3]: started with pid [22249]
process[crazyswarm_teleop-4]: started with pid [22253]
[ INFO] [1649325122.240838265]: Wait for services...
[ INFO] [1649325122.241209979]: waitForService: Service [/emergency] has not been advertised, waiting...
[ INFO] [1649325122.262309412]: waitForService: Service [/emergency] is now available.
[ INFO] [1649325122.263177828]: Manager ready.
[ INFO] [1649325122.266615338]: libmotioncapture connecting to vrpn at hostname '192.168.0.100' - might block indefinitely if unreachable!
[ WARN] [1649325122.269440772]: Couldn't set gain on joystick force feedback: Bad file descriptor
[ INFO] [1649325122.269882146]: Opened joystick: /dev/input/js0 (Microsoft X-Box 360 pad). deadzone_: 0.050000.
check_vrpn_cookie(): VRPN Note: minor version number doesn't match: (prefer 'vrpn: ver. 07.35', got 'vrpn: ver. 07.33  0').  This is not normally a problem.
ch: 1
[ INFO] [1649325122.430852885]: Parsed crazyflies.yaml successfully.
[ INFO] [1649325122.430920751]: Adding CF: cf4 (radio://0/80/2M/E7E7E7E704, cf4)...
terminate called after throwing an instance of 'std::runtime_error'
  what():  timeout
[crazyswarm_server-2] process has died [pid 22248, exit code -6, cmd /home/aarsh/crazyswarm/crazyswarm_cs/ros_ws/devel/lib/crazyswarm/crazyswarm_server __name:=crazyswarm_server __log:=/home/aarsh/.ros/log/5f8c5dc8-b658-11ec-a67f-5d216f884988/crazyswarm_server-2.log].
log file: /home/aarsh/.ros/log/5f8c5dc8-b658-11ec-a67f-5d216f884988/crazyswarm_server-2*.log

```

Any suggestions where I might have gone wrong?

Thanks.

**Comment by aarsht7** [comment, marked as answer]
I got the problem. 

Even I put `id: 4`, in configuration file it consider it as `04` and the address for the radio it takes is `radio://0/80/2M/E7E7E7E704`  instead of ~~`radio://0/80/2M/E7E7E7E74`~~   `radio://0/80/2M/E7E7E7E7E4` To change this behavior refer [this line in crazyswarm_server.cpp](https://github.com/USC-ACTLab/crazyswarm/blob/master/ros_ws/src/crazyswarm/src/crazyswarm_server.cpp#L1169) and change it to the `sstr << std::hex << id;` which now works perfectly.

**Reply by jpreiss** [reply]
this is intended behavior. Crazyswarm expects ids of the form `0xE7E7E7E7__` where the last byte is specified by the id field. Your desired address zero-extended to 5 bytes is `0x0E7E7E7E74`, which does not fit this template.

**Reply by aarsht7** [reply]
Sorry I did a typo. I meant `radio://0/80/2M/E7E7E7E7E4` the outof the box address is `0xE7E7E7E7E7` and I am using `0xE7E7E7E7E4`
